src/push_api_clientpy/platformclient.py

```diff
@@ -103,11 +107,26 @@ class BatchUpdateDocuments:
     addOrUpdate: list[Document]
     delete: list[BatchDelete]
 
+@dataclass
+class BackoffOptions:
+    retry_after: int = DEFAULT_RETRY_AFTER
+    max_retries: int = DEFAULT_MAX_RETRIES
+    time_multiple: int = DEFAULT_TIME_MULTIPLE
+
 
 class PlatformClient:
-    def __init__(self, apikey: str, organizationid: str):
+    def __init__(self, apikey: str, organizationid: str, backoff_options: BackoffOptions, session = requests.Session()):
         self.apikey = apikey
         self.organizationid = organizationid
+        self.backoff_options = backoff_options
+
+        self.retries = Retry(status=self.backoff_options.max_retries,
+                        status_forcelist=[429],
+                        backoff_factor=self.backoff_options.time_multiple,
+                        backoff_jitter=self.backoff_options.retry_after
+                        )
+        session.mount('http://', HTTPAdapter(max_retries=self.retries))
```

Comment:
Curious, what does this do? 
Should it be `http://` or `https://`?

Reply:
_**Whoops**_, yes it needs to be `https`! Otherwise the retry configuration won't take affect 😅